applications/FSIApplication/python_scripts/partitioned_fsi_base_solver.py

```diff
@@ -253,72 +211,75 @@ def SaveRestart(self):
         pass
 
     def SolveSolutionStep(self):
-        ## Solvers initialization
-        self.InitializeSolutionStep()
-
-        ## Solvers predict
-        self.Predict()
+        ## Safe ward to avoid coupling if fluid problem is not initialized
+        if not self.fluid_solver._TimeBufferIsInitialized():
+            return True
 
-        ## Compute mesh prediction ##
-        if (self.double_faced_structure):
-            self._ComputeMeshPredictionDoubleFaced()
-        else:
-            self._ComputeMeshPredictionSingleFaced()
-        KratosMultiphysics.Logger.PrintInfo("","\tMesh prediction computed.")
+        ## Initialize residual
+        dis_residual_norm = self._ComputeInitialResidual()
 
-        ## Non-Linear interface coupling iteration ##
+        ## FSI nonlinear iteration loop
         nl_it = 0
         is_converged = False
-        while not is_converged:
-            nl_it += 1
-            KratosMultiphysics.Logger.PrintInfo("","\tFSI non-linear iteration = ", nl_it)
-
-            self.fluid_solver.main_model_part.ProcessInfo[KratosMultiphysics.CONVERGENCE_ACCELERATOR_ITERATION] = nl_it
-            self.structure_solver.main_model_part.ProcessInfo[KratosMultiphysics.CONVERGENCE_ACCELERATOR_ITERATION] = nl_it
+        while (nl_it < self.max_nl_it):
+            # Check convergence
+            if self._CheckFSIConvergence(nl_it, dis_residual_norm):
+                KratosMultiphysics.Logger.PrintInfo('PartitionedFSIBaseSolver', 'FSI non-linear iteration convergence achieved in {0} iterations.'.format(nl_it))
+                is_converged = True
+                break
 
+            # Perform the displacement convergence accelerator update
             self._GetConvergenceAccelerator().InitializeNonLinearIteration()
+            if not self._GetConvergenceAccelerator().IsBlockNewton():
+                self._GetFSICouplingInterfaceStructure().Update()
+            else:
+                self._GetFSICouplingInterfaceStructure().UpdateDisplacement()
 
-            # Solve the mesh problem as well as the fluid problem
-            self._SolveMeshAndFluid()
+            # Update the structure interface position
+            self._GetFSICouplingInterfaceStructure().UpdatePosition(KratosMultiphysics.RELAXED_DISPLACEMENT)
```

Comment:
also here, is the indentation correct?

Reply:
I think it is. For each non-linear iteration you update the position of the coupling interface with the corrected solution (in this case stored in the `RELAXED_DISPLACEMENT` variable).